File: programming/project-learning/javascript/algorithms/simple-binary-search-tree.js
// Define an array of data representing nodes in the binary search tree
const data = ["A", "+", "B", "*", "C", "/", "D", "-", "E"];

// Define arrays representing the left and right children of each node
const left = [null, 0, null, 2, null, 1, null, 6, null];
const right = [null, 3, null, 4, null, 7, null, 8, null];

// Define the root node index
const root = 5;

// Define a function P to perform a post-order traversal of the binary search tree
function P(T) {
    // Recursively traverse the left subtree if it exists
    if (left[T] !== null) {
        P(left[T]);
    }
    // Recursively traverse the right subtree if it exists
    if (right[T] !== null) {
        P(right[T]);
    }
    // Print the data of the current node
    console.log(data[T]);
}

// Start the post-order traversal from the root node
P(root);

File: programming/project-learning/perl/algorithms/simple-binary-search-tree.pl
use strict;
use warnings;

# Array representing the data stored in the binary search tree nodes
my @data = ("A", "+", "B", "*", "C", "/", "D", "-", "E");

# Arrays representing the left and right children of each node
my @left = (undef, 0, undef, 2, undef, 1, undef, 6, undef);
my @right = (undef, 3, undef, 4, undef, 7, undef, 8, undef);

# Index of the root node in the @data array
my $root = 5;

# Subroutine to perform a post-order traversal of the binary search tree
sub P {
    my ($T) = @_;
    P($left[$T]) if defined $left[$T];  # Recursively traverse the left subtree
    P($right[$T]) if defined $right[$T];  # Recursively traverse the right subtree
    print $data[$T], "\n";  # Print the data of the current node
}

# Start the post-order traversal from the root node
P($root);